src/dialogxml/keycodes.cpp

```diff
@@ -100,6 +100,10 @@ unsigned char applyShift(unsigned char c){
 		'~', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O',
 		'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '{', '|', '}', '~',
 	};
+	if (c<' ')
+		return ' ';
+	if (c>=0x7f)
+		return c;
```

Comment:
Just noting that I already applied this fix myself, in a slightly different way.